server/controllers/stock/reports/stock/entry_purchase_receipt.js

```diff
@@ -66,16 +88,77 @@ async function stockEntryPurchaseReceipt(documentUuid, session, options) {
     document_uuid         : line.document_uuid,
     document_reference    : line.document_reference,
     purchase_reference    : line.purchase_reference,
+
+    exchangeRate          : exRate,
+    p_currency_id         : line.currency_id, // PO currency
+    p_shipping_handling   : line.shipping_handling, // Shipping+handling cost of full order (in PO currency)
+    p_full_cost           : line.cost, // Cost of full order not including shipping+handling (in PO currency)
+
     p_note                : line.note,
-    p_cost                : line.cost,
     p_date                : line.purchase_date,
     p_method              : line.payment_method,
     supplier_display_name : line.supplier_display_name,
     barcode               : barcode.generate(key, line.document_uuid),
     voucher_reference     : voucherReference,
     autoStockAccountingEnabled,
   };
+
+  // Set up flag for handlebars
+  data.not_in_enterprise_currency = session.enterprise.currency_id !== data.details.p_currency_id;
+
+  // // ??? TEMPORARY DIAGNOSITIC CODE
+  // console.log('====================================================================');
+  // console.log("DATA: ", data);
+  // console.log("PO LOTS: ", poLots);
+
+  // Add the original PO unit cost and quantity for each lot
+  let sumCost = 0.0;
+  let delivTotalQuantity = 0;
+  rows.forEach(row => {
+    // Get the article quantity and unit cost from the original PO
+    const poLot = poLots.find(p => p.inventory_uuid === row.inventory_uuid);
+    row.p_quantity = poLot.quantity;
+    row.p_unit_cost = poLot.unit_price;
+
+    // Unit cost without shipping+handling (in PO currency)
+    row.unit_cost_base = row.p_unit_cost;
+
+    // Compute actual unit_cost of the delivery including S&H (in PO currency)
+    // Note: row.unit_cost is from the stock_movement table.  It is in enterprise
+    //      currency and may include the distributed shipping+handling costs.
+    row.unit_cost_deliv = row.unit_cost * exRate;
+
+    // Compute the total cost for this article in the delivery (in PO currency)
+    // (May include distributed shipping+handling costs)
+    row.cost_deliv = row.unit_cost_deliv * row.quantity;
+
+    sumCost += row.cost_deliv;
+    delivTotalQuantity += row.quantity;
+  });
+
+  // // ??? TEMPORARY DIAGNOSITIC CODE
+  // console.log("ROWS: ", rows);
+
+  // Compute the total cost of all delivered articles (in PO currency)
+  // (This also includes distributed shipping+handling costs)
+  data.details.total_cost_deliv = sumCost;
+
+  // Compute the total cost of the original PO including shipping+handling (in PO currency)
+  data.details.p_total_cost = data.details.p_full_cost + data.details.p_shipping_handling;
+  data.details.p_total_cost_ent = data.details.p_total_cost / exRate; // Enterprise currency
```

Comment:
In the posting journal/general ledger, we call the enterprise currency version `debit_equiv` and `credit_equiv`.  For consistency, I propose you call this `p_total_cost_equiv` as well.

Reply:
Done